### PR TITLE
Ethaccount interface

### PR DESCRIPTION
**Pull request by Solpatium**

### CODE REVIEW

**Comment by bartekryba** [comment]
For now when executing a transaction, metamask popup will appear twice - first to estimate fee, second to actually execute the transaction. Transaction execution however doesn't use estimated fee yet, as it's not supported by devnet.

File: eip712-starknet-account/src/utils.ts [hunk 1]
```diff
@@ -21,25 +22,31 @@ export const parseSignature = (signature: string): Signature => {
   return [vStr, rLow, rHigh, sLow, sHigh];
 };
 
-export const computeAddress = (ethAddress: string, chainId: number) =>
+export const idForStarknetChain = (chain: StarknetChainId) => {
```
Comment (Solpatium) [inline eip712-starknet-account/src/utils.ts:25]: I think the values for chains are a little bit different, aren't they?

File: eip712-starknet-account/src/account/index.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    if (transactionsDetail.maxFee || transactionsDetail.maxFee === 0) {
+      maxFee_ = transactionsDetail.maxFee;
+    } else {
+      const estimatedFee = (await this.estimateFee(transactions, { nonce }))
+        .amount;
+
+      maxFee_ = estimatedFeeToMaxFee(estimatedFee).toString();
+    }
+
+    console.log(maxFee_);
```
Comment (Solpatium) [inline eip712-starknet-account/src/account/index.ts:81]: 👀

File: eip712-starknet-account/src/account/index.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  public async execute(
+    calls: Call | Call[],
+    abis: Abi[] | undefined = undefined,
+    transactionsDetail: InvocationsDetails = {}
+  ): Promise<AddTransactionResponse> {
+    const transactions = Array.isArray(calls) ? calls : [calls];
+    const nonce = toBN(transactionsDetail.nonce ?? (await this.getNonce()));
+    const maxFee: BigNumberish = '0';
+
+    // maxFee, not maxFee_, is used in add_transaction, because devnet doesn't support calling invoking functions
```
Comment (Solpatium) [inline eip712-starknet-account/src/account/index.ts:69]: Please add a TODO to change this later.

File: eip712-starknet-account/src/account/index.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+  /* Account deployment methods */
+
+  public async isDeployed(): Promise<boolean> {
+    const code = await this.getCode(this.address);
+    return !!code.bytecode.length;
+  }
+
+  public async deployAccount(): Promise<AddTransactionResponse> {
+    console.log('deployment starts', hexToDecimalString(implementationAddress));
```
Comment (Solpatium) [inline eip712-starknet-account/src/account/index.ts:116]: 👁️

**Pull request merged**